core/src/test/java/org/infinispan/container/offheap/OffHeapSingleNodeExpirationEvictionTest.java

```diff
@@ -64,4 +69,41 @@ protected EmbeddedCacheManager addClusterEnabledCacheManager(ConfigurationBuilde
       }
       return super.addClusterEnabledCacheManager(defaultConfig);
    }
+
+   public void testEnsureCorrectStorage() {
+      Cache<String, String> cache = cache(0);
+//      long beforeInsert = System.currentTimeMillis();
+      cache.put("k", "v");
+//      long afterInsert = System.currentTimeMillis();
+      CacheEntry<String, String> entry = cache.getAdvancedCache().getCacheEntry("k");
+      assertNotNull(entry);
+      long storedTime = TimeUnit.MINUTES.toMillis(10);
+      switch (expirationType) {
+         case MORTAL:
+            assertEquals(storedTime, entry.getLifespan());
+            assertEquals(-1, entry.getMaxIdle());
+//            assertBetweenTimes(beforeInsert, entry.getCreated(), afterInsert);
+            assertEquals(-1, entry.getLastUsed());
+            break;
+         case TRANSIENT:
+            assertEquals(-1, entry.getLifespan());
+            assertEquals(storedTime, entry.getMaxIdle());
+            assertEquals(-1, entry.getCreated());
+//            assertBetweenTimes(beforeInsert, entry.getLastUsed(), afterInsert);
+            break;
+         case TRANSIENT_MORTAL:
+            assertEquals(storedTime, entry.getLifespan());
+            assertEquals(storedTime, entry.getMaxIdle());
+//            assertBetweenTimes(beforeInsert, entry.getCreated(), afterInsert);
+//            assertBetweenTimes(beforeInsert, entry.getLastUsed(), afterInsert);
+            break;
+      }
+   }
+
+   // TODO: this is commented out because currently you can't retrieve the created or last used information from the cache
+   // See https://issues.jboss.org/browse/ISPN-7954 if this has been fixed.
```

Comment:
I thought of doing this instead, but I had I guessed incorrectly I had fixed the issue (I will change it over to verify for sure these are written properly). I really wish I could reproduce this issue on my box :(